opensearch-operator/api/v1/os_types.go

```diff
@@ -42,27 +42,25 @@ type OsGeneral struct {
 	ServiceName    string `json:"serviceName,omitempty"`
 }
 
-type OsNodes struct {
-	Replicas     int32  `json:"replicas,omitempty"`
-	DiskSize     int32  `json:"diskSize,omitempty"`
-	NodeSelector string `json:"nodeSelector,omitempty"`
-	Cpu          int32  `json:"cpu,omitempty"`
-	Memory       int32  `json:"memory,omitempty"`
-	Ingest       string `json:"ingest,omitempty"`
-	Jvm          string `json:"jvm,omitempty"`
-}
-
-type OsMasters struct {
-	/////////+kubebuilder:validation:Enum=3,5
+type OsNode struct {
+	Compenent    string `json:"compenent,omitempty"`
 	Replicas     int32  `json:"replicas,omitempty"`
 	DiskSize     int32  `json:"diskSize,omitempty"`
 	NodeSelector string `json:"nodeSelector,omitempty"`
 	Cpu          int32  `json:"cpu,omitempty"`
 	Memory       int32  `json:"memory,omitempty"`
 	Jvm          string `json:"jvm,omitempty"`
+	Ingest       string `json:"ingest,omitempty"`
+	Data         string `json:"data,omitempty"`
 }
 
-type OsConf struct {
+// OsConfMgmt defines which addiionals services will be deployed
+type OsConfMgmt struct {
+	Scaler     bool `json:"scaler,omitempty"`
+	AutoScaler bool `json:"autoScaler,omitempty"`
+	Monitoring bool `json:"monitoring,omitempty"`
+	Kibana     bool `json:"kibana,omitempty"`
```

Comment:
Didnt we say that kibana will have all relevant configuration in it like its port?